src/App.js

```diff
@@ -1,289 +1,182 @@
+          {props.title}
+        </button>
+      </div>
+    </>
   );
-}
+};
 
 const Search = (props) => {
+  const [contacts, setContacts] = React.useState([]);
+  const [searchTerm, setSearchTerm] = React.useState("");
 
-  const [ contacts, setContacts ] = useState([]);
-
-  const [searching, setSearch] = useState("");
-    useEffect( () => {
-      console.log("search changed", {searching})
-    }, [searching] );
-
-  // const [all, setAll] = useState(false);
-
-  const onSubmitSearch = async (e) => {
-      let search  = "%"+searching+"%";
-      let searchObject = { search };
-      e.preventDefault(); // prevents refreshing
-      try {
-        const body = searchObject ;
-        const response = await fetch("http://localhost:5000/lastname", {
-          method: "POST", // fetch makes GET request by default
-          headers: { "Content-Type": "application/json" },
-          body: JSON.stringify(body)
-        });
-        const jsonData = await response.json();
-        setContacts(jsonData);
-        console.log(jsonData);
-        
-        console.log(response);
-        console.log(searchObject);
-        setStuff(jsonData);
-
-        // window.location = "/"; // refresh after done sending
-      } catch (error) {
-        console.error(error.message);  
-      }
+  const queryContacts = (searchTerm) => async () => {
+    const url = new URL("/contacts", window.location.origin);
+    if (searchTerm !== undefined) {
+      url.search = new URLSearchParams({ lastName: searchTerm }).toString();
     }
-
-  const onSubmit0Search = async (e) => {
-      let search  = "%%";
-      let searchObject = { search };
-      e.preventDefault(); // prevents refreshing
-      try {
-        const body = searchObject ;
-        const response = await fetch("http://localhost:5000/lastname", {
-          method: "POST", // fetch makes GET request by default
-          headers: { "Content-Type": "application/json" },
-          body: JSON.stringify(body)
-        });
-        const jsonData = await response.json();
-        setContacts(jsonData);
-        console.log(jsonData);
-        
-        console.log(response);
-        console.log(searchObject);
-        setStuff(jsonData);
-        // window.location = "/"; // refresh after done sending
-      } catch (error) {
-        console.error(error.message);  
-      }
+    try {
+      setContacts(await (await fetch(url)).json());
+    } catch (error) {
+      console.error(error.message);
     }
-  
-  const [stuff, setStuff] = useState([]);
+  };
 
-  return(
+  return (
     <>
-    <h2>{props.title}</h2>
-    <div>
-      <label>{props.title}
-        <input 
-          type="text"
-          placeholder="search by last name"
-          onChange={ (e) => setSearch(e.target.value) }
-        />
-      </label>
-
-
-      <button
-      onClick = { onSubmitSearch }
+      <h2>{props.title}</h2>
+      <form
+        onSubmit={(e) => {
```

Comment:
Attach event handling to form submission, rather than button click for accessibility.